src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -94,6 +94,37 @@ class SourcesTree extends Component<Props, State> {
       debuggeeUrl,
       sources
     });
+    this.orderTree.bind(this);
+  }
+
+  orderTree() {
+    // extensions sources are stored temporarily for positioning them at last
+    const extensions = this.state.sourceTree.contents.filter(item => {
+      if (isUrlExtension(item.path)) {
+        this.state.sourceTree.contents.splice(
+          this.state.sourceTree.contents.indexOf(item),
+          1
+        );
+        return item;
+      }
+    });
+    // webpacks can be stored here and then appended in the sourceTree at
+    // desired position after main origin
+    const webpack = this.state.sourceTree.contents.filter(item => {
+      if (item.path === "webpack://") {
+        this.state.sourceTree.contents.splice(
+          this.state.sourceTree.contents.indexOf(item),
+          1
+        );
+        return item;
+      }
+    });
+    this.setState({
```

Comment:
i'm worried that this can be a performance hit. Would it be possible to limit this just the `addToTree` function in utils which might already have some sorting?

Reply:
Okay !

Reply:
@jasonLaster Actually, `addToTree` is traversing the tree, and appending the nodes to end of the tree.
So, If we implement the sort algorithm there, The time complexity will go to `n^2`  . However, Sorting before passing will make our work done in just 2 loops, i.e. `n` .
So I think the way I did is fine, though some changes like usage of for loops instead of filters can be used.
What do you suggest?